contracts/test/v0.8/foundry/dev/special/NoCancelVRFCoordinatorV2.t.sol

```diff
@@ -0,0 +1,78 @@
+pragma solidity ^0.8.0;
+
+import {BaseTest} from "../../BaseTest.t.sol";
+import {NoCancelVRFCoordinatorV2} from "../../../../../src/v0.8/dev/special/NoCancelVRFCoordinatorV2.sol";
+import {MockLinkToken as LinkToken} from "./MockLinkToken.sol";
+
+contract NoCancelVRFCoordinatorV2Setup is BaseTest {
+  NoCancelVRFCoordinatorV2 s_coordinator;
+
+  LinkToken LINK;
+
+  address internal LINK_ETH_FEED_ADDRESS = 0x2B4Ef6FAC5F6BE758508605bd87356776BfbfbA0;
+  address internal BLOCKHASH_STORE_ADDRESS = 0x032477835Bc60C777321e1e09F1Fe6cFB82e111C;
+
+  function setUp() public virtual override {
+    BaseTest.setUp();
+
+    LINK = new LinkToken();
+    s_coordinator = new NoCancelVRFCoordinatorV2(address(LINK), LINK_ETH_FEED_ADDRESS, BLOCKHASH_STORE_ADDRESS);
+  }
+}
+
+contract NoCancelVRFCoordinatorV2_cancelSubscription is NoCancelVRFCoordinatorV2Setup {
+  address internal subscriptionOwner = 0x054684DA2dCcf0e79E72DcAb1b5db42E71b66c8A;
+
+  event SubscriptionCanceled(uint64 indexed subId, address to, uint256 amount);
+  event SubscriptionCreated(uint64 indexed subId, address owner);
+  event SubscriptionFunded(uint64 indexed subId, uint256 oldBalance, uint256 newBalance);
+
+  function testCancelSubscription() public {
+    vm.expectRevert();
```

Comment:
nit: we can add the message of the revert to this assertion to be sure here.

Reply:
I tried `vm.expectRevert(<msg>)` but I couldn't get it to work, the expected value by forge seems to be `0x1f6a65b6`

```
[FAIL. Reason: Error != expected error: 0x1f6a65b6 != sub cancellation not allowed]
```

Reply:
Technically fine for it to revert on those reasons, but I imagine you are trying to test that even a valid subscription will revert once inside the implementation.

Reply:
Oh that's a good catch. I should create a subscription as the sub owner, and then try to cancel it from that address instead.

Reply:
Fixed in [ddcd19a](https://github.com/smartcontractkit/chainlink/pull/7928/commits/ddcd19ac95f0dd1257c16878bb648a5d6885f28a)